core/src/edu/asu/c3simulator/screens/TaskManagement.java

```diff
@@ -0,0 +1,273 @@
+package edu.asu.c3simulator.screens;
+
+import java.util.ArrayList;
+import java.util.Hashtable;
+
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.Button;
+import com.badlogic.gdx.scenes.scene2d.ui.ButtonGroup;
+import com.badlogic.gdx.scenes.scene2d.ui.CheckBox;
+import com.badlogic.gdx.scenes.scene2d.ui.ScrollPane;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.ui.VerticalGroup;
+import com.badlogic.gdx.scenes.scene2d.utils.Align;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+import com.badlogic.gdx.utils.Array;
+
+import edu.asu.c3simulator.util.Task;
+
+/**
+ * This class creates and displays the GUI component for managing tasks. The class
+ * separates available tasks into Action Based and Non-Action Based categories. The user
+ * can check all the tasks they want and press the add tasks button for tasks to be moved
+ * to the selected tasks column. Also, it is possible to remove tasks from the current
+ * tasks back to their categories by checking them and clicking the remove button.
+ * 
+ * @author Alyahya, Mohammed
+ */
```

Comment:
Where are the tasks loaded from? Can I add new tasks programatically, or is the list set?